pinot-plugins/pinot-input-format/pinot-csv/src/main/java/org/apache/pinot/plugin/inputformat/csv/CSVRecordReader.java

```diff
@@ -103,6 +105,23 @@ public void init(File dataFile, @Nullable Set<String> fieldsToRead, @Nullable Re
     _recordExtractor.init(fieldsToRead, recordExtractorConfig);
   }
 
+  private void validateHeaderForDelimiter(char delimiter, String csvHeader, CSVFormat format)
+      throws IOException {
+    CSVParser parser = format.parse(RecordReaderUtils.getBufferedReader(_dataFile));
+    CSVRecord firstRecord = parser.getRecords().get(0);
+    if (recordHasMultipleValues(firstRecord) && delimiterNotPresentInHeader(delimiter, csvHeader)) {
```

Comment:
lets stick to using `parser.Iterator()` (preferrably the one already created in the `init`). `getRecords` is internally creating an arraylist with all the records from the file.
firstRecord = iterator.hasNext() ? iterator.next() : null;

Reply:
using the iterator will also protect us from case where file has no records

Reply:
Yes, this is much more efficient. `getRecords` is reading the entire file. Addressed.